src/config.c

```diff
@@ -2934,6 +2925,31 @@ standardConfig configs[] = {
     {NULL}
 };
 
+/* Initialize configs to their default values and create the runtime
+ * configuration dictionary. */
+void initConfigValues() {
+    configs = dictCreate(&sdsHashDictType);
+    dictExpand(configs, sizeof(static_configs) / sizeof(standardConfig));
+    for (standardConfig *config = static_configs; config->name != NULL; config++) {
+        if (config->interface.init) config->interface.init(config->data);
+        standardConfig *primary = zmalloc(sizeof(standardConfig));
+        memcpy(primary, config, sizeof(standardConfig));
+        primary->flags |= PREFERRED_CONFIG;
+        primary->alias = NULL;
+        /* Add the primary config to the dictionary. */
+        dictAdd(configs, sdsnew(primary->name), primary);
+        /* Alias is not used at runtime, stead we create
```

Comment:
I'm honestly not sure what your comment here was about, but I saw that it was a good idea to add duplication checking, so if that was your point than I did that.

Reply:
Sorry - I think I picked the wrong word but yeah the config duplication is a good call out too.

I was commenting on how the new implementation treats aliased configs as **standalone** configs, meaning that we will now create two config entries if a config is aliased and then handle them independently for updates. In its current form, this change would break aliases **silently**. On the other hand, the old implementation was able to arrive at the same config entry via either the real name or alias(es). 

I think one solution is to refcount the config entry and not create a new config entry for the alias; rather, bump up the refcount on the config entry created for the real name and then have the alias point to the same config entry. Then when deallocating the dict/hashtable, make sure to free the config entry only after the refcount drops to 0.